gandalf/src/main/java/io/github/btkelly/gandalf/network/BootstrapApi.java

```diff
@@ -78,6 +88,16 @@ public BootstrapApi(Context context, @Nullable OkHttpClient okHttpClient, String
         } else {
             this.okHttpClient = okHttpClient;
         }
+
+        if (executor == null) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+                this.executor = context.getMainExecutor();
+            } else {
+                this.executor = new HandlerExecutor(new Handler(context.getMainLooper()));
+            }
```

Comment:
TIL.

Reply:
Yeah it is a simpler pattern and I needed to be able to inject a custom executor for unit tests. Also, the `getMainExecutor` seems like the newer pattern for running things on the main thread from outside of a UI context.